genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/SetUpJobAction.java

```diff
@@ -310,12 +313,41 @@ private void addEntitiesFilesToManifest(
         }
     }
 
+
+    private Map<String, String> createAdditionalEnvironmentMap(
+        final File jobDirectory,
+        final JobSpecification jobSpec
+    ) {
+        final Map<String, String> envMap = Maps.newHashMap();
```

Comment:
You could also use `ImmutableMaps.Builder<String, String> envMap = ImmutableMaps.builder()` which might be nicer than wrapping your said map in the collections unmodifiable interface

Reply:
Didn't know about that builder, will change